simespresso/io_QE/espresso_class.py

```
@@ -0,0 +1,815 @@
+                    try:
+                        line = f.next()
+                    except StopIteration:
+                        print('EOF')
+                        break
+            if len(particle_list):
+                pc.add_particles(particle_list)
+                #  part_container.add_particle(p)
+                n = self.count_particles()
+                print('n_particles=' + str(n))
+        except StopIteration:
+            return ('EOF')
+
+        return pc
+
+    def count_particles(self):
+        n = 0
+        for particle in self.pc.iter_particles():
+            n += 1
+        return n
+
+    def start_qe(self, name_in, name_out, path_to_espresso='./pw.x',
+                 mpi=False, mpi_Nprocessors=2):
+        print('starting start_qe')
+        if path_to_espresso is None:
+            path_to_espresso = './pw.x '
+# I'll assume if no path given then pw.x is on path
+        if mpi:
+            command = 'mpirun -np ' + str(mpi_Nprocessors) + ' ' + \
+                      path_to_espresso + ' < ' + name_in + ' > ' + name_out
+        else:
+            command = path_to_espresso + ' < ' + name_in + ' > ' + name_out
+
+#        if not os.path.isfile(path_to_espresso):
+#            logging.warning(path_to_espresso + ' is not on path')
+#           this may be ok if pw.x is defined somewhere on the PATH
+#            return None
+
+        print('start_qe attempting to run: ' + command)
+# alternative would be to use subprocess.check_call()  -
+# however this would give the same info as the
+# try/except, while taking twice as long in the case of success, iiuc
+        try:
+            subprocess.check_call(
+                command, shell=True,
+                stdout=subprocess.PIPE).stdout.read()
+
+#            subprocess.Popen(command, shell=True,
+#                             stdout=subprocess.PIPE).stdout.read()
+        except:
+            e = sys.exc_info()[0]
+            print("<p>Error: %s</p>" % e)
```

Why are you swallowing the exception?